### PR TITLE
Add smk65/revf support

### PR DESCRIPTION
**Pull request by zvecr**
## Description

* Assumed master (and backwards compatible) for now
* Minimal changes to existing rev code
* Added community support
* Alignment to current standards where possible

**Note:** The revF board I have for testing, has no RGB at all so I have left that as a TODO for someone that does.

Questions up for discussion...
- [ ] Should we split out revA, which would modify revB futher, without having hardware to test against?
- [ ] Should we target develop and just migrate action macros?
- [ ] Some revA/B keyboards do not have RGB enabled. Keeping the root level keymaps folder to maintain backwards compatibility causes potential issues on enabling those features by default (settings were already all over the place anyway). Should we default to off to bodge round?

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by tzarc — APPROVED** [review]

File: keyboards/lfkeyboards/smk65/keymaps/default/rules.mk [hunk 1]
```diff
@@ -1,26 +1,11 @@
 # Build Options
-#   change to "no" to disable the options, or define them in the Makefile in
-#   the appropriate keymap folder that will get included automatically
+#   change yes to no to disable
 #
+AUDIO_ENABLE = no              # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/lfkeyboards/smk65/keymaps/default/rules.mk:4]:
```suggestion
AUDIO_ENABLE = no              # Audio output
```

File: keyboards/lfkeyboards/smk65/keymaps/default/rules.mk [hunk 2]
```diff
@@ -1,26 +1,11 @@
...
+RGBLIGHT_ENABLE = no           # Disable RGB underlight
```
Comment (fauxpark) [inline keyboards/lfkeyboards/smk65/keymaps/default/rules.mk:5]:
```suggestion
RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
```

File: keyboards/lfkeyboards/smk65/keymaps/iso/rules.mk [hunk 3]
```diff
@@ -1,26 +1,11 @@
...
-MIDI_ENABLE = no                # MIDI controls
-AUDIO_ENABLE = no               # Audio output on port C6
-UNICODE_ENABLE = no             # Unicode
-BLUETOOTH_ENABLE = no           # Disable Bluetooth with the Adafruit EZ-Key HID
-RGBLIGHT_ENABLE = no          	# Disable RGB underlight
-RGBLIGHT_CUSTOM_DRIVER = yes    # RGB code is implemented in lefkeyboards, not WS2812
-SLEEP_LED_ENABLE = yes          # Breathing sleep LED during USB suspend
-TAP_DANCE_ENABLE = no
+AUDIO_ENABLE = no              # Audio output on port C6
+RGBLIGHT_ENABLE = no           # Disable RGB underlight
```
Comment (fauxpark) [inline keyboards/lfkeyboards/smk65/keymaps/iso/rules.mk:5]:
```suggestion
AUDIO_ENABLE = no              # Audio output
RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
```

File: keyboards/lfkeyboards/smk65/revb/rules.mk [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BACKLIGHT_ENABLE = no           # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no            # Enable keyboard RGB underglow
+RGBLIGHT_CUSTOM_DRIVER = yes    # RGB code is implemented in lefkeyboards, not WS2812
+AUDIO_ENABLE = no               # Audio output
+
+# Extra source files for IS3731 lighting
+SRC = TWIlib.c issi.c lighting.c
```
Comment (fauxpark) [inline keyboards/lfkeyboards/smk65/revb/rules.mk:16]:
```suggestion
SRC += TWIlib.c issi.c lighting.c
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]
Can get rid of the kb.h stuff and `cbi`/`sbi`, otherwise LGTM

**Review by drashna — APPROVED** [review]

**Pull request merged**